mlir/lib/Dialect/Rock/Transforms/ThreadwiseGemmLowering.cpp

```diff
@@ -203,60 +202,24 @@ struct XdlopsGemmV2RewritePattern : public OpConversionPattern<XdlopsGemmV2Op> {
     auto imms = mfmaGroup.getImms();
     int64_t nResultVectors = imms.size();
     Type argType = mfmaGroup.getArgType();
+    int64_t nRepeats = mfmaGroup.getNRepeats();
 
     MfmaInsnAttr mfmaAttr = mfmaGroup.getInsnAttr();
 
     int64_t mfmaNonKDim = mfmaAttr.mfmaNonKDim;
-    int64_t inputSpansPerMfmaIn = mfmaAttr.inputSpansPerMfmaIn;
-    int64_t blocksInOutRegs = mfmaAttr.blocksInOutRegs;
     int64_t k_base = mfmaAttr.k_base;
 
-    bool IsKReduction = (blocksInOutRegs == 1) && (inputSpansPerMfmaIn > 1);
-
-    Value matrixA = adaptor.getMatrixA();
-    Value matrixB = adaptor.getMatrixB();
-    Value matrixC = adaptor.getMatrixC();
-    auto matrixAType = matrixA.getType().cast<MemRefType>();
-    auto matrixBType = matrixB.getType().cast<MemRefType>();
+    Value bufferA = adaptor.getMatrixA();
```

Comment:
If you `TypedValue<MemRefType>` here, you don't need the `cast<>()` later down.

Reply:
Yes, this is leftover that should be get rid of, will remove.